src/main/java/org/ihiw/management/repository/UploadRepository.java

```diff
@@ -21,9 +21,11 @@
 public interface UploadRepository extends JpaRepository<Upload, Long> {
     List<Upload> findByCreatedByIn(List<IhiwUser> users);
     List<Upload> findByFileName(String filename);
+    
+    @Query("select upload from Upload upload where upload.createdBy in ?1 and upload.parentUpload.id is null")
```

Comment:
I'd suggest to create separate versions of these methods, because from the outside it's not apparent, you're just getting parent uploads with all of these methods. Not sure if there are already dependencies on the old method logic, that now will break - but certainly it's a risk for future method usages.

Reply:
Agreed. As far as I know there are no dependencies on the old methods, it's only the REST endpoints that are accessing these.

I'm renaming these three methods (and the intermediate layers) like this:
findParentlessByCreatedByIn
findParentlessByUsersAndProjects
findAll -> findParentless